mlir/lib/Dialect/Rock/Transforms/FoldTranspose.cpp

```diff
@@ -39,6 +41,199 @@ using namespace mlir;
 using namespace mlir::rock;
 
 namespace {
+
+// This rewrite will rewrite the linalg IO that has view like-ops surrounding
+// them to be consumed by the linalg operation itself adjusting the indexing
+// maps to faithfully represent them.
+struct InlineViewLikeOperandsLinalgRewritePattern
+    : public OpRewritePattern<linalg::GenericOp> {
+  using OpRewritePattern<linalg::GenericOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg::GenericOp laGeneric,
+                                PatternRewriter &rewriter) const override;
+};
+
+// This function will take strides calculated using reassociation of dimensions
+// that are modified by view-like ops and recompute the affine expr from src
+// view to dest view.
+static void getDelinearizedAffineExpr(mlir::ArrayRef<int64_t> strides,
```

Comment:
I'm confused by the naming in this function. I'd expect, for instance, a 3x4x5 memref to have `shapes = [3, 4, 5]` and strides equal to `[20, 5, 1]`, so these shapes vs strides comparisons make no sense to me

Reply:
Except that those are association subset of the shapes and the strides. So unless all the dimensions are being collapsed to 1d tensor, one will not be able to do infer one by the other.

I ll add more docs and probably rename shapes --> associatedDimSizes strides --> associatedDimStrides